src/main/java/com/google/firebase/messaging/FirebaseMessaging.java

```diff
@@ -256,7 +256,7 @@ private void handleSendHttpError(HttpResponseException e) throws FirebaseMessagi
     try {
       JsonParser parser = jsonFactory.createJsonParser(e.getContent());
       parser.parseAndClose(response);
-    } catch (IOException ignored) {
+    } catch (IOException | NullPointerException ignored) {
```

Comment:
Handling NPEs is almost never a good idea. Can this be avoided by checking `Strings.isNullOrEmpty(e.getContent())`, before calling the parser? That makes the intent more clear as well.

Reply:
I've addressed this now.